Understanding Blockchains Interoperability via Cosmos Network (#6984)
section-engineering-education/engineering-education: content/articles/understanding-blockchains-interoperability-via-cosmos-network/index.md

#### The networking layer
In this layer, the consensus process is involved. The nodes that are referred to as [non-validator nodes](https://loomx.io/developers/en/non-validator-node.html#validator-and-non-validator-nodes) contribute to the process.

The messages conveyed between the networking layers contain various data ranging from blocks, proposals, and even consensus votes. The [validator nodes](https://learn.radixdlt.com/article/what-is-a-validator-node) work alongside the non-validator nodes. This allows efficient interaction between the nodes, and network peers can quickly access the latest messages and transactions.

#### The application layer
As discussed earlier, the work of the consensus layer is to perform nodes validation. In contrast, the networking layer transmits messages. The application layer then passes transactions to networking and consensus layers.

> In this layer, the transactions happening on the blockchain are determined. They are then forwarded to the consensus layer. The consensus and application layer communication happen through an [Application Blockchain Interface](https://blog.cosmos.network/abci-the-application-blockchain-interface-f1bd8278cdd7) (ABCI). This interface is not limited to a specific programming language. Applications developed to operate on the cosmos network can interact with the base layers of the protocol. This is with disregard of the programming language used.

Reviewer: ```suggestion
In this layer, the transactions happening on the blockchain are determined. They are then forwarded to the consensus layer. The consensus and application layer communication happens through an [Application Blockchain Interface](https://blog.cosmos.network/abci-the-application-blockchain-interface-f1bd8278cdd7) (ABCI). This interface is not limited to a specific programming language. Applications developed to operate on the cosmos network can interact with the base layers of the protocol. This is with disregard for the programming language used.
```

### Pillars of the cosmos network
#### Tendermint core
Since the cosmos network utilizes different technological innovations, the efficiency and security of the interoperable systems are assured. Some of these technological innovations include the [Tendermint core](https://tendermint.com/core/).

The Tendermint core and other vital components form an important building block of the cosmos hub in the cosmos network. It includes the [Byzantine Fault Tolerance](https://www.fool.com/investing/stock-market/market-sectors/financials/cryptocurrency-stocks/byzantine-fault-tolerance) feature in its network architecture. The feature handles the regeneration of applications using the existing blockchain networks in the cosmos environment. This means it assists the users to switch to other existing blockchain networks.

For instance, an ABCI application can assist in deploying the Ethereum codebases on the cosmos network. This will ensure that the challenges of transferring [EVM](https://ethereum.org/en/developers/docs/evm/) to the cosmos are dealt with.

#### IBC protocol

> This [protocol](https://ibcprotocol.org/) handles interoperability between different blockchains in the cosmos network. It also sets up the interaction between blockchains through the [zone and hub model](https://www.preethikasireddy.com/post/how-does-cosmos-work-how-does-it-compare-to-bitcoin-and-ethereum-part-2). Hubs are used to manage different zones. There is indirect interaction between the zones through the use of IBC. The reason for indirect interaction is to ensure that the scalability of the blockchain is not affected.

Reviewer: The [Inter‑Blockchain Communication Protocol (IBC)](https://ibcprotocol.org/) handles interoperability between different blockchains in the cosmos network. It also sets up the interaction between blockchains through the [zone and hub model](https://www.preethikasireddy.com/post/how-does-cosmos-work-how-does-it-compare-to-bitcoin-and-ethereum-part-2). Hubs are used to manage different zones. There is indirect interaction between the zones through the use of IBC. The reason for indirect interaction is to ensure that the scalability of the blockchain is not affected.

#### Cosmos SDK
[Cosmos SDK](https://v1.cosmos.network/sdk) is another building block of the cosmos network. It significantly improves the innovation around the cosmos network. Developers do not have to develop applications from scratch. They can instead utilize the [cosmos PoS module](https://runtimeverification.com/blog/cosmos-modules-documentation/). Developers can also perform their customizations based on their needs and preferences.

> Since Cosmos SDK is modular in design, making blockchains easy to deploy. Developers utilize the cosmos SDK to develop blockchain applications. These applications include [Binance Chain](https://www.binance.com/en), [Terra](https://www.terra.money/), and [Kava](https://www.kava.io/).

Reviewer: ```suggestion
Cosmos SDK is modular in design, making blockchains easy to deploy. Developers utilize the cosmos SDK to develop blockchain applications. These applications include [Binance Chain](https://www.binance.com/en), [Terra](https://www.terra.money/), and [Kava](https://www.kava.io/).
```

### Popularly known projects built on cosmos network
#### Sifchain
It is a multichannel [DEX](https://www.coinbase.com/learn/crypto-basics/what-is-a-dex) developed using the cosmos SDK. This channel chose the cosmos network because it processes transactions fast at minimal costs. It also offers better performance and has been used in popularly used blockchains. 

[Sifchain](https://sifchain.finance/) can be integrated with Ethereum and cosmos networks to offer the users the advantages of each chain. It also ensured that the liquidity of the chains had been pooled together.

#### Band protocol
The [protocol](https://bandprotocol.com/bandchain) uses web APIs to extract the data. The data is then transferred to other blockchain smart contracts. It has also unlocked a variety of cross-chain functionalities that developers can utilize. The band protocol has partnered with Terra and [Mirror](https://www.mirror.finance/) protocol as their top providers.

#### pSTAKE

> Staked assets are not that liquid in the blockchain. This has not been easy for crypto-investors in years. [pStake](https://pstake.finance/) finance came to change the phenomena. They introduced a liquid staking protocol which is used in their cosmos environment. The protocol uses a [dual token model](https://phemex.com/academy/what-is-the-dual-token-model) whereby stakers can use their assets to issue one to one pegged representatives. It then prospects existing investment opportunities available in the [DeFi](https://www.coinbase.com/learn/crypto-basics/what-is-defi) ecosystem.

Reviewer: ```suggestion
Staked assets are not that liquid in the blockchain. This has not been easy for crypto-investors in years. [pStake](https://pstake.finance/) finance came to change the phenomena. They introduced a liquid staking protocol which is used in their cosmos environment. The protocol uses a [dual token model](https://phemex.com/academy/what-is-the-dual-token-model) whereby stakers can use their assets to issue one-to-one pegged representatives. It then prospects existing investment opportunities available in the [DeFi](https://www.coinbase.com/learn/crypto-basics/what-is-defi) ecosystem.
```
